sdcflows/interfaces/fmap.py

```diff
@@ -483,21 +578,50 @@ def phdiff2fmap(in_file, delta_te, newpath=None):
 
 
     """
-    import math
     import numpy as np
     import nibabel as nb
     from nipype.utils.filemanip import fname_presuffix
     #  GYROMAG_RATIO_H_PROTON_MHZ = 42.576
 
     out_file = fname_presuffix(in_file, suffix='_fmap', newpath=newpath)
     image = nb.load(in_file)
-    data = (image.get_data().astype(np.float32) / (2. * math.pi * delta_te))
+    data = (image.get_data().astype(np.float32) / (2. * np.pi * delta_te))
     nii = nb.Nifti1Image(data, image.affine, image.header)
     nii.set_data_dtype(np.float32)
     nii.to_filename(out_file)
     return out_file
 
 
+def rescale_phase_image(phase_data):
+    """
+    Ensure that phase images are in a usable range for unwrapping.
+
+    From the `FUGUE User guide
+    <https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/FUGUE/Guide>`__:
+
+        | If you have separate phase volumes that are in integer format then do:
+
+        .. code-block:: shell
+
+            fslmaths orig_phase0 -mul 3.14159 -div 2048 phase0_rad -odt float
+            fslmaths orig_phase1 -mul 3.14159 -div 2048 phase1_rad -odt float
+
+        | Note that the value of 2048 needs to be adjusted for each different
+        | site/scanner/sequence in order to be correct. The final range of the
+        | phase0_rad image should be approximately 0 to 6.28. If this is not the
+        | case then this scaling is wrong. If you have separate phase volumes are
+        | not in integer format, you must still check that the units are in radians,
+        | and if not scale them appropriately using fslmaths.
+
+    ..
+
+    """
+    imax = phase_data.max()
```

Comment:
There is already a function for this (https://github.com/niflows/nipype1-workflows/blob/361776115d5dc83b091fe1f8d15f6509c894faae/package/niflow/nipype1/workflows/dmri/fsl/utils.py#L625-L658).

If that doesn't work well or needs revision, please unify both. BTW, I see that you don't recenter to zero (i.e., from -pi to pi), I believe that was required by PRELUDE.

Reply:
[FUGUE guide](https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/FUGUE/Guide#Step_2_-_Getting_.28wrapped.29_phase_in_radians) specifies that range of values of phase images should be 0 to 6.28 before PRELUDE.

I tried recentering just in case, but the results don't look much better.